### PR TITLE
SWC-6321: Increase react-query stale time defaults

### PR DESCRIPTION
**Pull request by nickgros**
- react-query default stale time increased from 1m to 5m
- Table query stale time set to 30m
- default cacheTime increased from 5m to 30m.

### CODE REVIEW

File: src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ import {
 import { entityQueryKeys } from './queryKeys'
 
 const sharedQueryDefaults = {
+  staleTime: 1000 * 60 * 30, // 30 minutes
   refetchOnWindowFocus: false,
```
Comment (nickgros) [inline src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts:27]:
```suggestion
```

File: src/lib/utils/SynapseContext.tsx [hunk 2]
```diff
@@ -11,7 +11,8 @@ import { ThemeOptions } from '@mui/material'
 export const defaultQueryClientConfig: QueryClientConfig = {
   defaultOptions: {
     queries: {
-      staleTime: 60 * 1000, // 60s
+      staleTime: 1000 * 60 * 5, // 5 min
+      cacheTime: 1000 * 60 * 30, // 30 min
       retry: false, // SynapseClient knows which queries to retry
```
Comment (nickgros) [inline src/lib/utils/SynapseContext.tsx:16]:
```suggestion
      retry: false, // SynapseClient knows which queries to retry
      refetchOnWindowFocus: true,
```

File: src/lib/utils/SynapseContext.tsx [hunk 3]
```diff
@@ -11,8 +11,10 @@ import { ThemeOptions } from '@mui/material'
 export const defaultQueryClientConfig: QueryClientConfig = {
   defaultOptions: {
     queries: {
-      staleTime: 60 * 1000, // 60s
+      staleTime: 1000 * 60 * 5, // 5 min
+      cacheTime: 1000 * 60 * 30, // 30 min
```
Comment (nickgros) [inline src/lib/utils/SynapseContext.tsx:15]: So data will be considered "fresh" (won't be auto-refetched) for 5 minutes, and stale data could appear before being immediately refetched for 30 minutes.

File: src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ import {
 import { entityQueryKeys } from './queryKeys'
 
 const sharedQueryDefaults = {
-  refetchOnWindowFocus: false,
+  staleTime: 1000 * 60 * 30, // 30 minutes
```
Comment (nickgros) [inline src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts:26]: For table queries, max out the stale time to match the cache time, i.e. don't auto-refetch data for 30 minutes

File: src/lib/utils/SynapseContext.tsx [hunk 5]
```diff
@@ -11,8 +11,10 @@ import { ThemeOptions } from '@mui/material'
...
       retry: false, // SynapseClient knows which queries to retry
+      refetchOnWindowFocus: true,
```
Comment (nickgros) [inline src/lib/utils/SynapseContext.tsx:17]:
```suggestion
      refetchOnWindowFocus: false,
```

File: src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts (hunk 4 above)
Comment (jay-hodgson) [reply src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts:26]: Does this mean that after a Table is updated in SWC the user will not see the change for 30 minutes?

Comment (nickgros) [reply src/lib/utils/hooks/SynapseAPI/entity/useGetQueryResultBundle.ts:26]: They will see the change if they refresh the page (unless we move the cache to localStorage or sessionStorage), or if we invalidate the cache. The SWC EntityPresenter currently clears data for all entities from the cache when we fire the EntityUpdatedEvent.

**Pull request merged**